students/duanez2021/lesson04/mailroom_rev02.py

```diff
@@ -0,0 +1,170 @@
+
+def resolve_donor_name(db, s):
+    ret_name = s
+    for i, j in enumerate(db):
+        if s.lower() in db[i][0].lower():
+            ret_name = db[i][0]
+    return ret_name
+
+def add_donor_amt(db, donor_name, amt):
+    for i, j in enumerate(db):
+        if donor_name.lower() in db[i][0].lower():
+            db[i][1].append(float(amt))
+
+def prep_individual_email():
+    while True:
+        thankyou_to_name = input(thankyou_2_prompt)
+        while thankyou_to_name.lower() == "list":
+            list_donors(donor_db)
+            thankyou_to_name = input(thankyou_2_prompt)
+        if thankyou_to_name.lower() == "exit":
+            break
+        if not check_donor_name(donor_db, thankyou_to_name):
+            print("This is a new donor:")
+            amt = input(thankyou_for_amt_prompt)
+            add_new_donor(donor_db, thankyou_to_name, amt)
+        else:
+            amt = input(thankyou_for_amt_prompt)
+            add_donor_amt(donor_db, thankyou_to_name, amt)
+        # update data with new amount
+        send_email(thankyou_to_name)
+        # print("Send email to " + thankyou_to_name + ' for ' + str(amt))
+    return False
+
+def send_email(donor_name):
+    donor = resolve_donor_name(donor_db, donor_name)
+    print('Creating emails to {}.'.format(donor))
+    # for a single name, create an email file for each of their donations
+    for i, j in enumerate(donor_db):
+        if donor in j:
+            idx = i
+            n = get_number_of_donations(donor_db,donor)
+            for x in range(n):
+                template = 'Dear {:}, \n\n' '\t\tThank you for your very kind donation of ${:,.2f}.'
+                template += '\n\n\t\tIt will be put to very good use.\n\n'
+                template += '\t\t\tSincerly,\n\n'
+                template += '\t\t\t\t-The Team.\n'
+                email = template.format(donor, float(donor_db[idx][1][x]))
+                # print(template.format(s, float(donor_db[idx][1][x])))
+                #print(email)
+                fname = donor + '_' + str(x + 1) + 'of' + str(n) + '.txt'
+                with open(fname, 'w') as f:
+                    f.write(email)
+                    f.close()
+    return True
+
+
+def send_email_to_all():
+    # for a single name, create an email file for each of their donations
+    for i, j in enumerate(donor_db):
+        s = j[0]
+        send_email(s)
+    return True
+
+
+def exit_program():
+    print("Bye!")
+    sys.exit()  # exit the interactive script
+
+# main prompt switching dictionary
+main_prompt_dict = {1: prep_individual_email, 2: create_required_report, 3: send_email_to_all, 4: exit_program}
+
+def main():
+    while True:
+        response = input(new_prompt)
+        func = main_prompt_dict.get(int(response), lambda: print("Invalid Entry"))
+        func()
+
+if __name__ == "__main__":
+    # don't forget this block to guard against your code running automatically if this module is imported
+    main()
+# commit
+#
+
```

Comment:
@duanez2021 you'll need to convert donor_db to a dictionary and most of your code will change as you will be working with dictionary now.